### PR TITLE
fix(video): set video element if video id does not match element id

### PR DESCRIPTION
**Pull request by cmugla**
This is a fix for a bug found in the work here: https://github.com/Frameio/web-components/pull/1626

where the element is still animating into the frame, therefore the element does not exist quite yet, which results in a stale media interface. This adds another check for that scenario and sets the element.

### CODE REVIEW

File: src/video/video.js [hunk 1]
```diff
@@ -107,21 +117,34 @@ export default (
...
-            if (lastProps.videoId !== this.props.videoId) {
+
+            const hasIdChanged = lastProps.videoId !== this.props.videoId;
+            const videoElIdDoesNotMatchVideoId =
+                this.videoEl.id !== this.props.videoId;
+
+            if (hasIdChanged) {
+                this.unbindEvents();
+                this.getElementAndUpdateState(this.props.videoId);
+            } else if (videoElIdDoesNotMatchVideoId) {
```
Comment (cmugla) [inline src/video/video.js]: most of this is linting! And this is the only change

File: src/video/video.js [hunk 2]
```diff
@@ -113,15 +123,26 @@ export default (
...
-            if (lastProps.videoId !== this.props.videoId) {
+
+            const hasIdChanged = lastProps.videoId !== this.props.videoId;
+            const videoElIdDoesNotMatchVideoId =
+                this.videoEl && this.videoEl.id !== this.props.videoId;
+
+            if (hasIdChanged) {
+                this.unbindEvents();
+                this.getElementAndUpdateState(this.props.videoId);
+            } else if (videoElIdDoesNotMatchVideoId) {
```
Comment (snsullivan) [inline src/video/video.js]: could this be `if (hasIdChanged || videoElIdDoesNotMatchVideoId) { ... }`

**Review by snsullivan — APPROVED** [review]
code lgtm!

File: src/video/video.js (hunk 2 above)
Comment (cmugla) [reply src/video/video.js]: 🤦 doh yes

**Pull request merged**